### PR TITLE
Added python testing for secondary network interface

### PR DESCRIPTION
**Pull request by DejinChen**
Added python test for secondary network interface, REF to test plan https://github.com/CHIP-Specifications/chip-test-plans/pull/4218.

### CODE REVIEW

File: src/python_testing/TC_CNET_1_4.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+# test-runner-runs: run1
+# test-runner-run/run1/app: ${ALL_CLUSTERS_APP}
+# test-runner-run/run1/factoryreset: True
+# test-runner-run/run1/quiet: True
+# test-runner-run/run1/app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+# test-runner-run/run1/script-args: --storage-path admin_storage.json --commissioning-method on-network --discriminator 1234 --passcode 20202021 --PICS src/app/tests/suites/certification/ci-pics-values --trace-to json:${TRACE_TEST_JSON}.json --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+
+
+
+class TC_CNET_1_4(MatterBaseTest):
```
Comment (tcarmelveilleux) [inline src/python_testing/TC_CNET_1_4.py:33]: Please see comments on https://github.com/CHIP-Specifications/chip-test-plans/pull/4218

File: src/python_testing/TC_CNET_1_4.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        self.step(1)
+
+        cnet = Clusters.NetworkCommissioning
+        afeam = cnet.Attributes.FeatureMap
+
+        self.step(2)
+        # Read FeatureMap attribute with wildcard endpoint
+        feature_map_results = await self.default_controller.ReadAttribute(self.dut_node_id, [(afeam)], fabricFiltered=True)
+
+        NetworkNum = len(feature_map_results)
```
Comment (chshu) [inline src/python_testing/TC_CNET_1_4.py:63]: nit: use `NumNetworkCommissioning` in the script

**Review by chshu — APPROVED** [review]

File: src/python_testing/TC_CNET_1_4.py (hunk 1 above)
Comment (chshu) [reply src/python_testing/TC_CNET_1_4.py:32]: Addressed the comments in the test plan PR, the test script is also updated accordingly.

File: src/python_testing/TC_CNET_1_4.py (hunk 2 above)
Comment (DejinChen) [reply src/python_testing/TC_CNET_1_4.py]: Done

File: src/python_testing/TC_CNET_1_4.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    async def test_TC_CNET_1_4(self):
+        # Commissioning is already done
+        self.step(1)
+
+        cnet = Clusters.NetworkCommissioning
+        afeam = cnet.Attributes.FeatureMap
+
+        self.step(2)
+        # Read FeatureMap attribute with wildcard endpoint
+        feature_map_results = await self.default_controller.ReadAttribute(self.dut_node_id, [(afeam)], fabricFiltered=True)
```
Comment (ReneJosefsen) [inline src/python_testing/TC_CNET_1_4.py:66]:
```suggestion
        self.step(2)
        # Read FeatureMap attribute with wildcard endpoint
        feature_map_results = await self.default_controller.ReadAttribute(self.dut_node_id, [(Clusters.NetworkCommissioning.Attributes.FeatureMap)], fabricFiltered=True)
```

Is there any reason for not just adding it directly in the read?

File: src/python_testing/TC_CNET_1_4.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        if NumNetworkCommissioning == 0:
+            logging.info('No endpoint has Network Commissioning Cluster, skipping remaining steps')
+            self.skip_all_remaining_steps(3)
+            return
+
+        self.step(3)
+        cdesc = Clusters.Descriptor
+        aser = cdesc.Attributes.ServerList
+
+        server_list = await self.read_single_attribute_check_success(cluster=Clusters.Descriptor, attribute=aser, endpoint=0)
```
Comment (ReneJosefsen) [inline src/python_testing/TC_CNET_1_4.py:78]:
```suggestion
        server_list = await self.read_single_attribute_check_success(cluster=Clusters.Descriptor, attribute=cdesc.Attributes.ServerList, endpoint=0)
```

File: src/python_testing/TC_CNET_1_4.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        if 49 not in server_list:
```
Comment (ReneJosefsen) [inline src/python_testing/TC_CNET_1_4.py:79]: Not that this is going to change, but I assume we have a better way to get the ID than hard code it 🤔

File: src/python_testing/TC_CNET_1_4.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        # Read FeatureMap attribute with wildcard endpoint
+        feature_map_results = await self.default_controller.ReadAttribute(self.dut_node_id, [(afeam)], fabricFiltered=True)
+
+        NumNetworkCommissioning = len(feature_map_results)
+        if NumNetworkCommissioning == 0:
+            logging.info('No endpoint has Network Commissioning Cluster, skipping remaining steps')
+            self.skip_all_remaining_steps(3)
+            return
+
+        self.step(3)
```
Comment (ReneJosefsen) [inline src/python_testing/TC_CNET_1_4.py:74]: So, given we just did a wildcard read for the feature map attribute in Network Commissioning Cluster, should it be possible to check that response to see if there is a Network Commissioning Cluster on EP0? 🤔

Comment (DejinChen) [reply src/python_testing/TC_CNET_1_4.py:73]: check the response instead.

**Review by cecille — APPROVED** [review]

**Pull request merged**